miden-tx/src/error.rs

```diff
@@ -138,3 +138,152 @@ impl fmt::Display for DataStoreError {
 
 #[cfg(feature = "std")]
 impl std::error::Error for DataStoreError {}
+
+// KERNEL ASSERTION ERROR
+// ================================================================================================
+
+const ERR_FAUCET_RESERVED_DATA_SLOT: u32 = 131072;
+const ERR_ACCT_MUST_BE_A_FAUCET: u32 = 131073;
+const ERR_P2ID_WRONG_NUMBER_OF_INPUTS: u32 = 131074;
+const ERR_P2ID_TARGET_ACCT_MISMATCH: u32 = 131075;
+const ERR_P2IDR_WRONG_NUMBER_OF_INPUTS: u32 = 131076;
+const ERR_P2IDR_RECLAIM_ACCT_IS_NOT_SENDER: u32 = 131077;
+const ERR_P2IDR_RECLAIM_HEIGHT_NOT_REACHED: u32 = 131078;
+const ERR_SWAP_WRONG_NUMBER_OF_INPUTS: u32 = 131079;
+const ERR_SWAP_WRONG_NUMBER_OF_ASSETS: u32 = 131080;
+const ERR_KERNEL_TX_NONCE_DID_NOT_INCREASE: u32 = 131081;
+const ERR_KERNEL_ASSET_MISMATCH: u32 = 131082;
```

Comment:
This happens in the epilogue, right? 

Maybe we call it `ERR_EPILOGUE_ASSETS_DONT_ADD_UP`?